src/TrancheWrapper.sol

```diff
@@ -0,0 +1,236 @@
+                              CONSTRUCTOR
+    ////////////////////////////////////////////////////////////// */
+
+    /**
+     * @notice The constructor for a TrancheWrapper.
+     * @param asset_ The underlying ERC20 asset.
+     * @param name_ The name of the ERC4626 token.
+     * @param symbol_ The symbol of the ERC4626 token.
+     * @param TRANCHE_ The Tranche contract address.
+     */
+    constructor(ERC20 asset_, string memory name_, string memory symbol_, address TRANCHE_)
+        ERC4626(asset_, name_, symbol_)
+    {
+        TRANCHE = TRANCHE_;
+        LENDING_POOL = address(ITranche(TRANCHE_).LENDING_POOL());
+    }
+
+    /* //////////////////////////////////////////////////////////////
+                                FUNCTIONS
+    ////////////////////////////////////////////////////////////// */
+
+    /**
+     * @notice Compliant with the standard ERC-4626 deposit implementation.
+     * @param assets The amount of assets of the underlying ERC20 token being deposited.
+     * @param receiver The address that receives the minted shares.
+     * @return shares The amount of shares minted.
+     * @dev This contract does not directly transfer the underlying assets from the sender to the receiver.
+     * Instead it calls the deposit of the Lending Pool which calls the transferFrom of the underlying assets.
+     * Hence the sender should not give this contract an allowance to transfer the underlying asset but the Lending Pool instead.
+     */
+    function deposit(uint256 assets, address receiver) public override returns (uint256 shares) {
+        asset.safeTransferFrom(msg.sender, address(this), assets);
+
+        // Approval has to be given to the Lending Pool for the deposit in the Tranche.
+        asset.safeApprove(LENDING_POOL, assets);
+
+        shares = ERC4626(TRANCHE).deposit(assets, address(this));
+
+        _mint(receiver, shares);
+
+        emit Deposit(msg.sender, receiver, assets, shares);
+    }
+
+    /**
+     * @notice Compliant with the standard ERC-4626 mint implementation.
+     * @param shares The amount of shares minted.
+     * @param receiver The address that receives the minted shares.
+     * @return assets The amount of assets of the underlying ERC20 token being deposited.
+     * @dev This contract does not directly transfer the underlying assets from the sender to the receiver.
+     * Instead it calls the deposit of the Lending Pool which calls the transferFrom of the underlying assets.
+     * Hence the sender should not give this contract an allowance to transfer the underlying asset but the Lending Pool instead.
+     */
+    function mint(uint256 shares, address receiver) public override returns (uint256 assets) {
+        assets = ITranche(TRANCHE).previewMintAndSync(shares);
+
+        asset.safeTransferFrom(msg.sender, address(this), assets);
+
+        asset.safeApprove(LENDING_POOL, assets);
+
+        ERC4626(TRANCHE).deposit(assets, address(this));
```

Comment:
Not more gas efficient to call `deposit `here instead of `mint` (as I thought initially), as the same flow is trigerred via  `previewMintAndSync` and `previewDepositAndSync`. Thus cleaner to call `mint` as well on the tranche.